src/main/java/com/flansmod/client/ClientRenderHooks.java

```diff
@@ -582,12 +582,21 @@ public void renderPlayer(RenderPlayerEvent.Pre event)
 		}
 	}
 	
-	public void cameraSetup(CameraSetup event)
+public void cameraSetup(CameraSetup event)
 	{
 		if(mc.thePlayer.ridingEntity instanceof IControllable)
 		{
 			IControllable cont = (IControllable)mc.thePlayer.ridingEntity;
-			event.roll = interpolateRotation(cont.getPrevPlayerRoll(), cont.getPlayerRoll(), (float)event.renderPartialTicks);
+			float roll = interpolateRotation(cont.getPrevPlayerRoll(), cont.getPlayerRoll(), (float)event.renderPartialTicks);
+			if(((EntitySeat)mc.thePlayer.ridingEntity).driveable != null){
+			EntityDriveable ent = ((EntitySeat)mc.thePlayer.ridingEntity).driveable;
+			
+			if(ent.getDriveableType().canRoll){
```

Comment:
Actually this is to disable the camera from rolling in ground vehicles. If roll is enabled for a vehicle the camera would ordinarily roll with it, however this is rather jerky and makes it pretty difficult to see where you are going. This disables camera rolling for vehicles with the canRoll, but leaves everything else untouched (I.e. Planes).